### PR TITLE
codegen: Remove protocol test skipping for supported features.

### PR DESCRIPTION
**Pull request by jasdel**
Updates the SDK's code generation of protocol test to no longer skip
tests for features that are supported by the SDK.

Regenerates SDK API clients for https://github.com/aws/smithy-go/pull/360 and previously skipped protocol tests

### CODE REVIEW

**Comment by jasdel** [comment]
Unit test failing because presigned URL behavior needs to deviate from regular API behavior and no include or remove the auto set content-type for operations with streaming input.

**Comment by skmcgrail** [comment]
This change in behavior I believe is unexpected.

```
--- FAIL: TestPutObject_PresignURL (0.05s)
    --- FAIL: TestPutObject_PresignURL/nil_body_with_content-length (0.00s)
        presign_test.go:277: expected signed header to be map[Content-Length:[100] Host:[mock-bucket.s3.us-west-2.amazonaws.com]], got map[Content-Length:[100] Content-Type:[application/octet-stream] Host:[mock-bucket.s3.us-west-2.amazonaws.com]], 
             diff :   http.Header{
              	"Content-Length": {"100"},
            + 	"Content-Type":   {"application/octet-stream"},
              	"Host":           {"mock-bucket.s3.us-west-2.amazonaws.com"},
              }
```

File: codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AwsHttpPresignURLClientGenerator.java [hunk 1]
```diff
@@ -21,6 +21,7 @@
 import java.util.Set;
 import java.util.TreeMap;
 import java.util.TreeSet;
+import javax.swing.plaf.synth.SynthMenuItemUI;
```
Comment (skmcgrail) [inline codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AwsHttpPresignURLClientGenerator.java:24]: Is this unused? You probably didn't want to pull this in regardless.

Comment (jasdel) [reply codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AwsHttpPresignURLClientGenerator.java:24]: Thanks, not quire sure where that came from, but yeah should not be added!

**Review by skmcgrail — APPROVED** [review]

**Pull request merged**